SCIM server doc updates (#2501)
FusionAuth/fusionauth-site: site/docs/v1/tech/shared/_scim-limits.adoc

The current SCIM implementation does not support bulk operations.

Bulk API operations are available via link:/docs/v1/tech/apis/[FusionAuth APIs], but are not currently supported via SCIM compatible endpoints. If this functionality is required, please file a https://github.com/fusionauth/fusionauth-issues/issues[GitHub issue] explaining your use case.

The list operation is limited to 10,000 users if you are using the Elasticsearch search engine. This https://github.com/FusionAuth/fusionauth-issues/issues/494[GitHub issue] discusses this limitation.

You cannot create a FusionAuth Registration for a user provisioned with SCIM directly during the provisioning process. See link:/docs/v1/tech/core-concepts/scim#adding-registrations[Adding Registrations] for workarounds.

> While it is not enforced, or a hard limitation, it is recommended to configure at most one SCIM server per tenant. While you can have multiple SCIM servers configured to provision users in one FusionAuth tenant, at the end of the day you only have one user data store in each tenant. Conflicts with creation, updates and deletion could happen; they won't be automatically resolved by FusionAuth. If you need to use multiple SCIM servers, ensure your clients can handle such conflicts.

Reviewer: From the doc:
> FusionAuth can consume SCIM formatted requests and act as the provisioning server in a SCIM client/server environment.

I think the user of "server" here is referring to a customer's server, but that server is acting as the _SCIM client_. Can we use "client" here instead of "server"?

> ensure your clients can handle such conflicts.

Does a _SCIM client_ also retrieve and consume data from the _SCIM server_? Or does "clients" here refer to anything that queries FusionAuth for user data?

> Okay, okay, you want multiple SCIM clients to provision users into a single FusionAuth tenant. Let's talk about the complexities. Imagine you have an Okta SCIM client provisioning users into a FusionAuth tenant and an Azure AD SCIM client provisioning users into the same tenant. Issues with this approach include:

Reviewer: The list doesn't format correctly
![image](https://github.com/FusionAuth/fusionauth-site/assets/3409780/88e9d622-f14d-458e-af48-746ef7d14cb7)

In my experience something like this works:

```
Lead in paragraph:
+
 * Bullet 1
 * Bullet 2
 * Bullet 3
```

An extra line before the list starts and put a space before each bullet point.

* The user and group lambdas would have to be smart enough to handle the mapping for both sources. You'd include logic differentiating between an user from Okta and a user from Azure AD, and handle the fields differently.You don't get information from the incoming SCIM request about the source, but you might be able to examine an email domain or a marker attribute to determine where the user was coming from.

> * Collisions of users will not be handled by FusionAuth. In particular, if the Okta SCIM client created 'richard@example.com' in the tenant, then later the Azure AD SCIM client tried to create 'dan@example.com', there would be a failure in the second case. One workaround would be for the lambda to be smart enough to resolve the collision. It might adding a suffix or prefix to make the email address unique. 

Reviewer: ```suggestion
* User collisions will not be handled by FusionAuth. For example, if the Okta SCIM client created `richard@example.com` in the tenant, then later the Azure AD SCIM client tried to create `richard@example.com`, there would be a failure in the second case. One workaround would be for the lambda to resolve the collision, for example by adding a suffix or prefix to make the email address unique.
```

> * Update or deletion requests from the Okta SCIM client could modify users created by the Azure AD server and vice versa.

Reviewer: ```suggestion
* Update or deletion requests from the Okta SCIM client could modify users created by the Azure AD SCIM client and vice versa.
```

> * the bearer access token can be different, but must have the same lifetime, since that is configured at the Entity Type level.

Author: Expanded on why this is relevant.
